media_driver/agnostic/common/renderhal/renderhal_dsh.cpp

```diff
@@ -2276,10 +2276,15 @@ int32_t RenderHal_DSH_LoadCurbeData(
                 if (iCurbeSize > 0)
                 {
                     uint8_t* remainingCurbe = (uint8_t*)MOS_AllocAndZeroMemory(sizeof(uint8_t)*iCurbeSize);
-                    MHW_RENDERHAL_CHK_STATUS(pDynamicState->memoryBlock.AddData(
```

Comment:
because MHW_RENDERHAL_CHK_STATUS will jump tp finish with error status, would it be simpler to free ramainingCurne inside if(eStatus != MOS_STATUS_SUCCESS)? much less lines changed.